pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -148,8 +149,12 @@ private Calendar getCalendar(ZoneOffset offset) {
       return calCache;
     }
 
-    calCache = new GregorianCalendar(TimeZone.getTimeZone(offset));
-    calCacheZone = offset;
+    // normally we would use:
+    // calCache = new GregorianCalendar(TimeZone.getTimeZone(offset));
+    // But this seems to cause issues for some crazy offsets as returned by server for BC dates!
+    // final String tzid = (offset.getTotalSeconds() == 0) ? "UTC" : "GMT".concat(offset.getId());
+    calCacheZone = TimeZone.getDefault();
```

Comment:
This is only temporary, right? Now all tests fail! :-)

I think this method is used for parsing only, maybe to fix the "appendTimeZone" a separate method to get a clone of default calendar is needed?